packages/type-utils/src/getConstrainedTypeAtLocation.ts

```diff
@@ -18,3 +28,59 @@ export function getConstrainedTypeAtLocation(
 
   return constrained ?? nodeType;
 }
+
+export interface ConstraintTypeInfoBase {
+  constraintType: ts.Type | undefined;
+  isTypeParameter: boolean;
+  type: ts.Type;
+}
+
```

Comment:
[Style] This doesn't enforce much and isn't used anywhere outside the `extends`:

```suggestion
```

Reply:
fair enough. i thought this was sort of a required part of the idiom but yeah it can be left off 👍